### ISSUE TITLE
feat(groups): target instances by specifier type

### ISSUE THREAD

**Issue by moltar** [JamieMason/syncpack#163]
## Description

Our use case is a bit of an edge case, and is not a problem with syncpack. But I am wondering if _a_ solution could be introduced as a feature to deal with this.

A tool we use for repo management has a workflow that goes like this:

1. The user adds a dependency to the tool config (not using `npm install foo`)
2. Tool renders `package.json` file with the new dependency using `*` as a placeholder.

```json
"dependencies": {
  "my-package": "*"
}
```

3. Tool automatically runs `{pnpm,npm,yarn} install`, to install the `*` version, which is a SemVer way of saying "latest".
4. Once the dep is installed, then tool runs a function to get the actual version of the installed package. In essence, all it does is `require.resolve('my-package')` to get the package location and then reads `package.json` to get the version.
5. Finally, once the tool knows the _actual_ version of the package, it rewrites our `package.json` again, and sets the _actual_ version.

```json
"dependencies": {
  "my-package": "^1.2.3"
}
```

6. Tool runs install once again to update the lock files.

This can be also reproduced and visualized without a particular tool, with just `npm` and CLI:

```sh
## start a new package
npm init -y

## add `lodash@*` dep without installing it
npm pkg set 'dependencies.lodash'='*'

## runs install
npm i

## update `package.json` to list _actual_ version, instead of `*`
npm pkg set 'dependencies.lodash'=$(npm ls --depth=0 lodash --json | jq -r '.version')

## run install again, to update `package-lock.json`
npm i
```

The problem occurs because of the way we use `syncpack`. It runs as `postinstall` script. And there's no way to discriminate the first install from the second install. So it runs twice.

And when it runs the first time, the version is still `*`. Syncpack will set this version across the whole workspace (monorepo). Correctly so, of course, as it is ultimately the highest version, as far as SemVer is concerned.

But in our case, it is only transient, but it affects all other packages.

The temporary fix for us was to set `preferVersion` to `lowestSemver`, but this introduces other problems, that I won't get into, as they are not really important here.

## Suggested Solution

I'd like to be able to ignore certain SemVers, notably `*`.

## Help Needed

N/A

**Comment by moltar** [comment]
I like the `specifierTypes` approach! 👍🏼 

> might you be better to set and check some environment variables to help decide when to run certain tasks or not

I did look into this, but did not find a way. The tool does not do this. And otherwise, there is no way to distinguish on `npm i` from another `npm i`.

My simplified example using CLI is almost literally all the tool does under the hood.

Imagine if my CLI example was wrapped into a single shell script, that you have no influence of.

The only changes I could make are to our environment, e.g. edit `package.json` in some way, or _not_ run `syncpack` in `postinstall`.

> Finally, if you or anyone reading this finds syncpack useful, please tell people about it – it's completely free and has been a ton of work, please help us spread the word – thanks!

Thank you for all your hard work, Jamie. I am certainly talking about the tool :)

**Comment by JamieMason** [comment]
Checking if the output of syncpack list contains a `*`? Yeah that could work. I would tweak that example slightly though as I think in the next version I might have changed it to output `help` instead of running `list` by default:

```diff
- check-versions && syncpack || true
+ check-versions && syncpack list || true
```

**Comment by moltar** [comment]
Right, my example was oversimplified, we run it with exact command and config file flag and all that 😁

**Comment by JamieMason** [comment]
Released in [12.2.0](https://github.com/JamieMason/syncpack/releases/tag/12.2.0).

👋 If anyone reading this finds syncpack useful, please tell people about it.
